curl/inc/leatherman/curl/client.hpp

```diff
@@ -108,6 +108,48 @@ namespace leatherman { namespace curl {
     };
 
     /**
+     * The exception for HTTP file downloads.
+     */
+    struct LEATHERMAN_CURL_EXPORT http_file_download_exception : http_exception
+    {
+        /**
+         * Constructs an http_file_download_exception.
+         * @param url The url that caused the exception.
+         * @param file_path The file that was meant to be downloaded
+         * @param message The exception message.
+         */
+        http_file_download_exception(std::string const &url, std::string const &file_path, std::string const &message) :
```

Comment:
The standard pattern for construction is to pass by value, then move into the object. You're mixing them by passing by reference, which is causing an extra copy.